src/core/load-fragment.c

```diff
@@ -4165,6 +4165,80 @@ int config_parse_protect_system(
 
 DEFINE_CONFIG_PARSE_ENUM(config_parse_exec_keyring_mode, exec_keyring_mode, ExecKeyringMode, "Failed to parse keyring mode");
 
+int config_parse_job_timeout_sec(
+                const char* unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        Unit *u = data;
+        usec_t usec;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(u);
+
+        r = parse_sec_fix_0(rvalue, &usec);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse JobTimeoutSec= parameter, ignoring: %s", rvalue);
+                return 0;
+        }
+
+        if (usec == USEC_INFINITY) {
```

Comment:
This should solve the immediate issue, but I had the same thought as @Werkov in https://github.com/systemd/systemd/issues/6402#issuecomment-333564159: shouldn't this `if` be dropped`? This patch restores compatibility for the specific case of timeout=infinity, but not for other values.